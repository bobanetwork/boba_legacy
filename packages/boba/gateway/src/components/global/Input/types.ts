import { ReactNode } from 'react'

export interface InputProps {
  placeholder: string
  type: string
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  value: any
  label?: ReactNode
}
