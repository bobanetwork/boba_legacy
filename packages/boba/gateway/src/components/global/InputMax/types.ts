export interface MaxInputTypes {
  max: number
  initialValue?: number
  onValueChange: (value: number) => void
}
