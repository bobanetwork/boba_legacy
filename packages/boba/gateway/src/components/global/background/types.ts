export enum BackgroundPosition {
  TOP = 'top',
  CENTER = 'center',
}

export interface BackgroundProps {
  position: BackgroundPosition
}

export const RoutesWithBackgroundPositionAtTop = ['/bridge', '/']
