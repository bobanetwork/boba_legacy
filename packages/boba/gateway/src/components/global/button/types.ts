import { ReactNode } from 'react'

type ButtonType = {
  disable?: boolean
  loading?: boolean
  small?: boolean
  outline?: boolean
  tiny?: boolean
  transparent?: boolean
}

export interface ButtonTypes extends ButtonType {
  label: ReactNode | string
  style?: Record<string, string>
  className?: string
  onClick?: () => void
}
