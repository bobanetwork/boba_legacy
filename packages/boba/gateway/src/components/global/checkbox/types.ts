export interface CheckboxWithLabelProps {
  label?: string
  checked?: boolean
  onChange: (isChecked: boolean) => void
}
