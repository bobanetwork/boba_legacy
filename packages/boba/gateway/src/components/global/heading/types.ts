import { ReactNode } from 'react'

export type VariantType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'

export interface HeadingProps {
  variant: VariantType
  color?: string
  children?: ReactNode
  className?: string
}

export interface HeadingStyleProps {
  size: string
  lineHeight: string
  fontWeight: number
}
