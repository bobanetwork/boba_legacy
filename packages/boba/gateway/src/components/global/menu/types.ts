import { ComponentType, ReactElement, ReactNode } from 'react'

export interface OptionProps {
  label?: string
  component?: ReactElement<any>
  onClick?: (d: any) => void
}

export interface MenuProps {
  name?: string
  label?: ReactNode | string
  children?: ReactNode
  options: Array<OptionProps>
}
