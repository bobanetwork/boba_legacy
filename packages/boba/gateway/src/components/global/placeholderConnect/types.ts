export interface PlaceholderConnectInterface {
  preloader?: any
  isLoading?: boolean
}
