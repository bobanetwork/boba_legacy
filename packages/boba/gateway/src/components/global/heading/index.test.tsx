import React from 'react'
import { render } from '@testing-library/react'
import { HeadingProps } from './types'
import CustomThemeProvider from 'themes'
import { Heading } from './'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'

const mockStore = configureStore()

const renderHeading = (props: HeadingProps) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'dark',
        },
      })}
    >
      <CustomThemeProvider>
        <Heading {...props}>Test Heading</Heading>
      </CustomThemeProvider>
    </Provider>
  )
}

describe('Heading', () => {
  test('should match snapshots h1', () => {
    const { asFragment } = renderHeading({
      variant: 'h1',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshots h2', () => {
    const { asFragment } = renderHeading({
      variant: 'h2',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshots h3', () => {
    const { asFragment } = renderHeading({
      variant: 'h3',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshots h4', () => {
    const { asFragment } = renderHeading({
      variant: 'h4',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshots h5', () => {
    const { asFragment } = renderHeading({
      variant: 'h5',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('should match snapshots h6', () => {
    const { asFragment } = renderHeading({
      variant: 'h6',
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
