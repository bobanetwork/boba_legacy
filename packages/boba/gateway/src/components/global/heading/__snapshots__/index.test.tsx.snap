// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading should match snapshots h1 1`] = `
<DocumentFragment>
  <h1
    class="sc-bgqQcB jYvPsJ"
  >
    Test Heading
  </h1>
</DocumentFragment>
`;

exports[`Heading should match snapshots h2 1`] = `
<DocumentFragment>
  <h2
    class="sc-bgqQcB hisnER"
  >
    Test Heading
  </h2>
</DocumentFragment>
`;

exports[`Heading should match snapshots h3 1`] = `
<DocumentFragment>
  <h3
    class="sc-bgqQcB hPrUKN"
  >
    Test Heading
  </h3>
</DocumentFragment>
`;

exports[`Heading should match snapshots h4 1`] = `
<DocumentFragment>
  <h4
    class="sc-bgqQcB bblAst"
  >
    Test Heading
  </h4>
</DocumentFragment>
`;

exports[`Heading should match snapshots h5 1`] = `
<DocumentFragment>
  <h5
    class="sc-bgqQcB jfkGcq"
  >
    Test Heading
  </h5>
</DocumentFragment>
`;

exports[`Heading should match snapshots h6 1`] = `
<DocumentFragment>
  <h6
    class="sc-bgqQcB jDQrGH"
  >
    Test Heading
  </h6>
</DocumentFragment>
`;
